agent/job_runner.go

```diff
@@ -149,32 +161,35 @@ type jobAPI interface {
 var _ jobRunner = (*JobRunner)(nil)
 
 // Initializes the job runner
-func NewJobRunner(l logger.Logger, scope *metrics.Scope, ag *api.AgentRegisterResponse, job *api.Job, apiClient APIClient, conf JobRunnerConfig) (jobRunner, error) {
-	runner := &JobRunner{
-		agent:     ag,
-		job:       job,
-		logger:    l,
+func NewJobRunner(conf JobRunnerConfig) (jobRunner, error) {
+	r := &JobRunner{
+		job:       conf.Job,
+		logger:    conf.Logger,
 		conf:      conf,
-		metrics:   scope,
-		apiClient: apiClient,
+		metrics:   conf.MetricsScope,
+		apiClient: conf.APIClient,
+	}
```

Comment:
I don't like the redundancy of having these available in two places, inside conf and outside conf.